src/oscar/apps/dashboard/reports/reports.py

```diff
@@ -34,8 +38,18 @@ def report_description(self):
                'end_date': date(self.end_date, 'DATE_FORMAT')
                }
 
+    def get_queryset(self):
+        if self.queryset is not None:
+            return self.queryset
+
+        if not self.model_class:
+            raise ValueError(
+                _("Please set model_class property for your report generator class or override get_queryset method.")
```

Comment:
Generally we don't translate developer error notices in Oscar (see other places where we use ValueError) - I think this makes sense as it's not an error that a site user should ever see, and adds more translation work.

Reply:
Also please adjust the grammar to "Please define a model_class property on your report generator class, or override the qet_queryset() method".

Reply:
Ok done.